cli/tools/coverage/ast_parser.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_ast::get_syntax;
+use deno_ast::MediaType;
+use deno_ast::ModuleSpecifier;
+use deno_ast::ParseDiagnostic;
+use deno_ast::ParsedSource;
+
+pub(crate) fn parse_program(
```

Comment:
Why do we need to parse an AST now? It seems we really only need the comments which the lexer provides?

Reply:
For the current implementation, there's no need to parse the AST if we can get comments from the Lexer. The current implementation parses the AST because it was largely copied/adapted from the [implementation in `deno_lint`](https://github.com/denoland/deno_lint/blob/f2a2aab27c3825403b538faeecb15f65f6179d9a/src/ignore_directives.rs), which does the same.

That said, I didn't consider using the lexer for comments in part because I had [syntax aware coverage ignore directives](https://github.com/denoland/deno/issues/26722) in mind while working on this. Since these require access to the AST, it didn't make sense to optimise away the AST parsing.

I'm happy to refactor to use the lexer if you prefer.